internal/Network/DNS/Types/Internal.hs

```diff
@@ -1330,6 +1330,35 @@ defaultResponse = DNSMessage {
   , additional = []
   }
 
+-- | A 'DNSMessage' template for NOTIFY queries for the given zone,
+-- as per RFC-1996.
+
+notifyMessage :: ByteString -> DNSMessage
+notifyMessage zoneName = DNSMessage {
+  header = DNSHeader {
+      identifier = 0
+    , flags = DNSFlags {
+             qOrR = QR_Query
+           , opcode = OP_NOTIFY
+           , authAnswer = True
+           , trunCation = False
+           , recDesired = True
+           , recAvailable = False
+           , rcode = NoErr
+           , authenData = False
+           , chkDisable = False
+      }
+    }
+    , ednsHeader = EDNSheader defaultEDNS
+    , question = [ Question { qname = zoneName
+                            , qtype = SOA
+                            }
+                 ]
+    , answer = []
```

Comment:
There should perhaps be an optional (normally SOA) ResourceRecord to all to the answer section. This helps the receiving system avoid unnecessary queries to its master if it is already at or ahead of the serial number in the provided SOA. NOTIFY without SOA is less efficient.

This suggests a signature of `notifyMessage :: ByteString -> [ResourceRecord] -> DNSMessage` or `notifyMessage :: ByteString -> DNSMessage -> Maybe ResourceRecord -> DNSMessage`. Probably the former for full generality. Callers that want to skip the SOA can just pass `[]` (else `Nothing`).

Alternatively, the second argument could be a `Maybe RData`, with the `ResourceRecord` constructed from the given zone, class `IN` and the TTL from the SOA. The result would be a `Maybe DNSMessage` with `Nothing` returned when the `RData` provided is not an `RD_SOA`.

Or some variation on the above.